pipe_anchorages/transforms/voyages_create.py

```diff
@@ -0,0 +1,62 @@
+import apache_beam as beam
+import datetime as dt
+import pytz
+
+def create_voyage(visits):
+    # first voyage start from 'we do not know' = None
+    visit_before = {2: None, 3: None, 4: None}
+
+    def min_confidence(v, c:int)->bool:
+        return v['confidence'] >= c
+
+    def has_duration(v1, v2):
+        return v1 == None or v1['end_timestamp'] < v2['start_timestamp']
+
+    for visit in visits:
+        for c in [2,3,4]:
+            if (visit and min_confidence(visit, c) and has_duration(visit_before[c],visit)) or (not visit and visit_before[c]):
+                yield build_voyage(c, visit_before[c], visit)
+                visit_before[c] = visit
+
+def build_voyage(trip_confidence, visit_before, visit_after):
```

Comment:
I would rename these to `start_visit` and `end_visit` here. You are building a voyage between these 2 visits, so it makes more sense to me.